src/daft-core/src/series/ops/if_else.rs

```diff
@@ -5,7 +5,10 @@ use common_error::DaftResult;
 
 impl Series {
     pub fn if_else(&self, other: &Series, predicate: &Series) -> DaftResult<Series> {
-        let (if_true, if_false) = match_types_on_series(self, other)?;
-        if_true.inner.if_else(&if_false, predicate)
+        let series = match_types_on_series(vec![self, other])?;
```

Comment:
also nit: rename `series` variable to `casted_series`? also whats the plural for series? is it just series? would be nice to know that the result of this could be multiple series

Reply:
Plural of series is just series unfortunately 😅, renamed though!